components/html-block/html-block.js

```diff
@@ -3,6 +3,43 @@ import { css, LitElement } from 'lit-element/lit-element.js';
 import { htmlBlockMathRenderer } from '../../helpers/mathjax.js';
 import { requestInstance } from '../../mixins/provider-mixin.js';
 
+let configuredIfrauClient;
+
+const getIfrauClient = async() => {
+	if (configuredIfrauClient === undefined && window.ifrauclient) {
+		configuredIfrauClient = await window.ifrauclient({
+			resizeFrame: false
+		}).connect();
+	}
+
+	return configuredIfrauClient;
+};
+
+let contextPromise;
+
+const getContext = () => {
+	if (contextPromise) return contextPromise;
+
+	contextPromise = new Promise(async resolve => { // eslint-disable-line no-async-promise-executor
```

Comment:
Seems like we should be able to avoid this by moving the ifrau bit out of this promise handler (i.e. move it above).

Reply:
Oh, you're right. I can do that in the editor too I think.